charts/osm/values.schema.json

```diff
@@ -258,24 +256,46 @@
-                    "title": "The certificateManager schema",
-                    "description": "The certificate manager osm-controller should use.",
-                    "pattern": "^(tresor|vault|cert-manager)$",
-                    "examples": [
-                        "tresor"
-                    ]
-                },
-                "serviceCertValidityDuration": {
-                    "$id": "#/properties/OpenServiceMesh/properties/serviceCertValidityDuration",
-                    "type": "string",
-                    "title": "The serviceCertValidityDuration schema",
-                    "description": "The service certificate validity duration.",
-                    "examples": [
-                        "24h"
-                    ]
+                "certificateProvider": {
+                    "$id": "#/properties/OpenServiceMesh/properties/certificateProvider",
+                    "type": "object",
+                    "title": "The certificate provider schema",
+                    "description": "Certificate provider configuration parameters",
+                    "required": [
+                        "kind",
+                        "serviceCertValidityDuration",
+                        "certKeyBitSize"
+                    ],
+                    "properties": {
+                        "kind": {
+                            "$id": "#/properties/OpenServiceMesh/properties/certificateProvider/properties/kind",
+                            "type": "string",
+                            "title": "The certificate provider kind schema",
+                            "description": "The certificate manager osm-controller should use.",
+                            "pattern": "^(tresor|vault|cert-manager)$",
+                            "examples": [
+                                "tresor"
+                            ]
+                        },
+                        "serviceCertValidityDuration": {
+                            "$id": "#/properties/OpenServiceMesh/properties/certificateProvider/properties/serviceCertValidityDuration",
+                            "type": "string",
+                            "title": "The serviceCertValidityDuration schema",
+                            "description": "The service certificate validity duration.",
+                            "examples": [
+                                "24h"
+                            ]
+                        },
+                        "certKeyBitSize": {
+                            "$id": "#/properties/OpenServiceMesh/properties/certificateProvider/properties/certKeyBitSize",
+                            "type": "integer",
+                            "title": "The certKeyBitSize schema",
+                            "description": "The key size for data plane certificates.",
+                            "examples": [
+                                2048
+                            ]
+                        }
```

Comment:
set `"additionalProperties": false` so that invalid keys are rejected for the root property.